buildpacks/ruby/CHANGELOG.md

```diff
@@ -7,7 +7,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
-- The buildpack now implements Buildpack API 0.10 instead of 0.9, and so requires `lifecycle` 0.17.x or newer. ([#283](https://github.com/heroku/buildpacks-ruby/pull/283/files#commit-suggestions))
+### Changed
+
+- The buildpack now implements Buildpack API 0.10 instead of 0.9, and so requires `lifecycle` 0.17.x or newer. ([#283](https://github.com/heroku/buildpacks-ruby/pull/283))
+
+### Added
+
+- Added support for Ubuntu 24.04 (and thus Heroku-24 / `heroku/builder:24`). ([#284](https://github.com/heroku/buildpacks-ruby/pull/284))
+
+### Fixed
```

Comment:
```suggestion
```

Reply:
I think that I would like these sections to already be in the CHANGELOG and then we just fill them out. That way you don't end up with two people adding a `Fixed` section in their PR and having extra conflicts to deal with. It's a bit more explicit as well, if someone is less familiar with keep a changelog format then they can tell at a glance that there were no "fixed" changes in this release. Not that useful now, but helpful for cases where there are no "changed" or "added" changes.

It could be more explicit with a `- None` bulletpoint under it, but that would lead to conflicts as it needs to be deleted.

I'm okay to take it out, but I'm suggesting we pre-populate this section when we cut a new release with the most common types of changes that we track.

Reply:
With that workflow, would you leave the empty sections in when you perform the actual release, or remove at that point? I don't like the idea of leaving empty sections in the actual release (given that there are 6 categories mentioned on https://keepachangelog.com/en/1.1.0/ and typically each release only has 1-2 categories used per release entry).

Reply:
I wrote a versioning RFC previously. I originally wanted the contents of the changelog to drive rev-ing the version.

I think we should strongly tie the tags we use in the CHANGELOG to versioning i.e. major, minor, patch. I would want those mapped to "keep a changelog" tags. Some are redundant or subsets. We can choose to only implement a subset of those tags by default and if we choose to extend later, then that's okay too.